src/main/scala/EmrSparkPlugin.scala

```diff
@@ -66,6 +63,8 @@ object EmrSparkPlugin extends AutoPlugin {
       settingKey[JobFlowInstancesConfig]("default JobFlowInstancesConfig")
     val sparkRunJobFlowRequest =
       settingKey[RunJobFlowRequest]("default RunJobFlowRequest")
+    val sparkS3SSEEnabled =
+      settingKey[Boolean]("Upload Spark Jar with server side encryption")
```

Comment:
For the flexibility of customizing `PutObjectRequest`, how if we provide a function like
```scala
val sparkS3PutObjectDecorator = settingKey[PutObjectRequest => PutObjectRequest]("...")
```
and let user apply the encryption by themselves in `build.sbt` like below?
```scala
import com.amazonaws.services.s3.model.ObjectMetadata
sparkS3PutObjectDecorator := { req =>
  val metadata = new ObjectMetadata()
  metadata.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION)
  req.withMetadata(metadata)
}
```

Reply:
awesome suggestion.